### DISCUSSION TITLE
Use Focus Visible for Tooltip

### DISCUSSION THREAD

**Discussion by nassimbenkirane** [chakra-ui/zag#1826] [Q&A] [answered]
It would be cool if the Tooltip machine used focus-visible rather than focus for opening the tooltip : 
Now it causes some weird UX opening the tooltip on some focus events not triggered by user intent

- After keyboard navigation out and back in a page

https://github.com/user-attachments/assets/468048a9-5cb3-4fc3-9c18-2000259e9f11

- After gaining back focus from closing a modal

https://github.com/user-attachments/assets/6dac9231-4e08-4797-a72c-ce57c2d20152

Is it something that could be improved in the future ?

**Comment by segunadebayo** [comment]
Yes, this is indeed a good UX improvement we can add.

**Reply by segunadebayo** [reply, marked as answer]
This is now implemented in Zag.js 🎉
